scripts/hfButler_cpp_clone.py

```diff
@@ -0,0 +1,378 @@
+            wn.getNode('csr.ctrl.chain').write(loop)
+            wn.getClient().dispatch()
+            
+        print('Fire!!!')
+        wn.fire()
+
+
+# -----------------------------------------------------------------------------
+@cli.command('flowmaster')
+@click.option('--src-sel', type=click.Choice(['gbt', 'wibtor']), help='Input source selection', default=None)
+@click.option('--sink-sel', type=click.Choice(['hits']+['link'+str(i) for i in range(5)]+['link-all']), help='Sink input selection', default=None)
+@click.pass_obj
+def flowmaster(obj, src_sel, sink_sel):
+
+    fmNode = obj.podNode.get_flowmaster_node()
+
+    src_map = {'gbt': 0, 'wibtor': 1}
+    sink_map = {'hits': (0, 0) }
+    sink_map.update({ 'link'+str(d): (1, d) for d in range(obj.mConfigInfo['n_links']) })
+    if src_sel:
+        if src_sel == 'gbt':
+            fmNode.set_source_gbt()
+        elif src_sel == 'wibtor':
+            fmNode.set_source_wtor()
+        else:
+            print("Invalid source")
+    if sink_sel is not None:
+        if sink_sel == 'hits':
+            fmNode.set_sink_hits()
+            fmNode.getClient().dispatch()
+        if (sink_sel[:4] == 'link'):
+            if sink_sel[4:] == '-all':
+                for l in range(obj.mConfigInfo['n_links']):
+                    fmNode.set_sink_link(l)               
+                    fmNode.getClient().dispatch()
+                    print(l)
+            if sink_sel[4] in [str(lnk) for lnk in range(obj.mConfigInfo['n_links'])]:
+                fmNode.set_sink_link(int(sink_sel[-1]))
+                fmNode.getClient().dispatch()
+# -----------------------------------------------------------------------------
+@cli.command('cr-if')
+@click.option('--on/--off', 'cr_on', help='Enable central-router interface block', default=None)
+@click.option('--drop-empty/--keep-empty', 'drop_empty', help='Drop empty hit packets', default=None)
+@click.pass_obj
+def cr_if(obj, cr_on, drop_empty):
+
+    crNode = obj.podNode.get_crif_node()
+    if cr_on is not None:
+        crNode.getNode('csr.ctrl.en').write(0x01)
+        crNode.getNode('csr.ctrl.drop_empty').write(0x01)
+        obj.mHW.dispatch()
+        
+    if drop_empty is not None:
+        crNode.getNode('csr.ctrl.drop_empty').write(0x01)
+        crNode.getClient.dispatch()
+
+    print('--- cr_if.csr ---')
+    regs = controls.get_child_registers(crNode.getNode("csr"))
+    printRegTable(regs, False)
+
+# -----------------------------------------------------------------------------
+@cli.command('capture-sink')
+@click.option('--path', type=click.Path(exists=False), help='Capture file path', default='b33_dump.txt')
+@click.option('--timeout', type=float, help='Capture timeout', default=10)
+@click.option('--drop-idles', is_flag=True, help='Drop idle words', default=None)
+@click.pass_obj
+def capture_sink(obj, path, timeout, drop_idles):
+
+    osNode = obj.podNode.get_output_sink_node()
+
+    print('Capturing axis32bsink data')
```

Comment:
not necessary immediately, but this should all be replaced with appropriate methods on OutputSinkNode

Reply:
Done reasonably I think. Please review.

Reply:
LGTM